src/passes/analysis/verilog.cpp

```diff
@@ -431,8 +570,7 @@ std::map<ConnMapKey, std::vector<ConnMapEntry>> build_connection_map(
         connection.first->getTopParent() == instance.second &&
         connection.first->getType()->isInput()) {
         SelectPath first_sel_path = connection.first->getSelectPath();
-        int index = 0;
-        if (first_sel_path.size() > 2) { index = std::stoi(first_sel_path[2]); }
+        std::vector<int> index = selPathToIndex(first_sel_path);
```

Comment:
just noticing that all these snippets look the same (below). perhaps an opportunity to merge the code paths.

Reply:
Yea, it's a bit annoying because it's 4 different variants of connections (either first to second, or second to first, with another two for self vs non self),  I think I might be able cook up a helper function that at least reduces the repetition, but we need the four cases.